pkgs/development/python-modules/black/default.nix

```diff
@@ -1,42 +1,71 @@
-{ stdenv, buildPythonPackage, fetchPypi, pythonOlder
-, attrs, click, toml, appdirs, aiohttp, aiohttp-cors
-, glibcLocales, typed-ast, pathspec, regex
-, setuptools_scm, pytest }:
+{ stdenv, lib
+, buildPythonPackage, fetchPypi, pythonOlder, setuptools_scm, pytestCheckHook
+, aiohttp
+, aiohttp-cors
+, appdirs
+, attrs
+, click
+, mypy-extensions
+, pathspec
+, regex
+, toml
+, typed-ast
+, typing-extensions }:
 
 buildPythonPackage rec {
   pname = "black";
-  version = "19.10b0";
+  version = "20.8b1";
 
   disabled = pythonOlder "3.6";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0f8mr0yzj78q1dx7v6ggbgfir2wv0n5z2shfbbvfdq7910xbgvf2";
+    sha256 = "1spv6sldp3mcxr740dh3ywp25lly9s8qlvs946fin44rl1x5a0hw";
   };
 
   nativeBuildInputs = [ setuptools_scm ];
-  checkInputs =  [ pytest glibcLocales ];
 
   # Necessary for the tests to pass on Darwin with sandbox enabled.
   # Black starts a local server and needs to bind a local address.
   __darwinAllowLocalNetworking = true;
 
-  # Don't know why these tests fails
-  # Disable test_expression_diff, because it fails on darwin
-  checkPhase = ''
-    LC_ALL="en_US.UTF-8" pytest \
-      --deselect tests/test_black.py::BlackTestCase::test_expression_diff \
-      --deselect tests/test_black.py::BlackTestCase::test_cache_multiple_files \
-      --deselect tests/test_black.py::BlackTestCase::test_failed_formatting_does_not_get_cached
+  checkInputs =  [ pytestCheckHook ];
+
+  preCheck = ''
+    export PATH="$PATH:$out/bin"
   '';
 
-  propagatedBuildInputs = [ attrs appdirs click toml aiohttp aiohttp-cors pathspec regex typed-ast ];
+  disabledTests = [
+    # Don't know why these tests fails
+    "test_cache_multiple_files"
+    "test_failed_formatting_does_not_get_cached"
```

Comment:
Could you maybe show a log of why these fail, so people can help? No real objection to disabling it though.

Reply:
Attached the log below. I don't know why the tests fail but they seem to fail this way consistently.

```
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /build/black-20.8b1
collected 150 items / 2 deselected / 148 selected                              

tests/test_black.py .........F..............................F........... [ 35%]
..................................................................xxx... [ 83%]
..................                                                       [ 95%]
tests/test_primer.py ......                                              [100%]

=================================== FAILURES ===================================
___________________ BlackTestCase.test_cache_multiple_files ____________________

self = <tests.test_black.BlackTestCase testMethod=test_cache_multiple_files>

    @event_loop()
    def test_cache_multiple_files(self) -> None:
        mode = DEFAULT_MODE
        with cache_dir() as workspace, patch(
            "black.ProcessPoolExecutor", new=ThreadPoolExecutor
        ):
            one = (workspace / "one.py").resolve()
            with one.open("w") as fobj:
                fobj.write("print('hello')")
            two = (workspace / "two.py").resolve()
            with two.open("w") as fobj:
                fobj.write("print('hello')")
            black.write_cache({}, [one], mode)
            self.invokeBlack([str(workspace)])
            with one.open("r") as fobj:
                self.assertEqual(fobj.read(), "print('hello')")
            with two.open("r") as fobj:
>               self.assertEqual(fobj.read(), 'print("hello")\n')
E               AssertionError: "print('hello')" != 'print("hello")\n'
E               - print('hello')
E               ?       ^     ^
E               + print("hello")
E               
E               ?       ^     ^ 
E               +

tests/test_black.py:1381: AssertionError
___________ BlackTestCase.test_failed_formatting_does_not_get_cached ___________

self = <tests.test_black.BlackTestCase testMethod=test_failed_formatting_does_not_get_cached>

    @event_loop()
    def test_failed_formatting_does_not_get_cached(self) -> None:
        mode = DEFAULT_MODE
        with cache_dir() as workspace, patch(
            "black.ProcessPoolExecutor", new=ThreadPoolExecutor
        ):
            failing = (workspace / "failing.py").resolve()
            with failing.open("w") as fobj:
                fobj.write("not actually python")
            clean = (workspace / "clean.py").resolve()
            with clean.open("w") as fobj:
                fobj.write('print("hello")\n')
>           self.invokeBlack([str(workspace)], exit_code=123)

tests/test_black.py:1456:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_black.py:196: in invokeBlack
    self.assertEqual(
E   AssertionError: 0 != 123 : Failed with args: ['--verbose', '--config', '/build/black-20.8b1/tests/empty.toml', '/build/tmpg7cs37d1']
E   stdout: ''
E   stderr: '/build/tmpg7cs37d1/clean.py ignored: matches the --exclude regular expression\n/build/tmpg7cs37d1/failing.py ignored: matches the --exclude regular expression\nNo Python files are present to be formatted. Nothing to do 😴\n'
E   exception: None
=============================== warnings summary ===============================
/nix/store/nlvjjvb9471dbpdj8fhdih0ak90cq9d9-python3.8-aiohttp-3.6.2/lib/python3.8/site-packages/aiohttp/helpers.py:107
  /nix/store/nlvjjvb9471dbpdj8fhdih0ak90cq9d9-python3.8-aiohttp-3.6.2/lib/python3.8/site-packages/aiohttp/helpers.py:107: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def noop(*args, **kwargs):  # type: ignore

tests/test_black.py: 12 tests with warnings
  /nix/store/nlvjjvb9471dbpdj8fhdih0ak90cq9d9-python3.8-aiohttp-3.6.2/lib/python3.8/site-packages/aiohttp/connector.py:964: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    hosts = await asyncio.shield(self._resolve_host(

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== short test summary info ============================
FAILED tests/test_black.py::BlackTestCase::test_cache_multiple_files - Assert...
FAILED tests/test_black.py::BlackTestCase::test_failed_formatting_does_not_get_cached
===== 2 failed, 143 passed, 2 deselected, 3 xfailed, 13 warnings in 29.16s =====
builder for '/nix/store/sjzrn2vmvw8b782n7pjsbdac1r2jmp7q-python3.8-black-20.8b1.drv' failed with exit code 1
error: build of '/nix/store/sjzrn2vmvw8b782n7pjsbdac1r2jmp7q-python3.8-black-20.8b1.drv' failed
```